### PR TITLE
Gromacs cray GPU + dynamic libs

### PR DESCRIPTION
**Pull request by gppezzi**

### CODE REVIEW

File: easybuild/easyblocks/g/gromacs.py [hunk 1]
```diff
@@ -97,7 +102,10 @@ def configure_step(self):
                 for libname in ['BLAS', 'LAPACK']:
                     lib_dir = os.getenv('%s_LIB_DIR' % libname)
                     libs = os.getenv('LIB%s' % libname)
-                    self.cfg.update('configopts', '-DGMX_%s_USER="-L%s %s"' % (libname, lib_dir, libs))
+                    if self.toolchain.name.startswith('Cray'):
```
Comment (boegel) [inline easybuild/easyblocks/g/gromacs.py]: please use:

```python
import easybuild.tools.toolchain as toolchain
...
    if self.toolchain.toolchain_family() == toolchain.CRAYPE:
```

File: easybuild/easyblocks/g/gromacs.py [hunk 2]
```diff
@@ -147,11 +155,19 @@ def install_step(self):
         # rather than trying to replicate the logic, we just figure out where the library was placed
 
         if LooseVersion(self.version) < LooseVersion('5.0'):
-            # libgmx.a or libgmx_mpi.a
-            libname = 'libgmx*.a'
+            if self.toolchain.options['dynamic']:
```
Comment (boegel) [inline easybuild/easyblocks/g/gromacs.py]: just determine the extension to use outside of this if/else block?

```python
if self.toolchain.options['dynamic']:
    libext = get_shared_lib_ext()
else:
    libext = 'a'

if LooseVersion(self.version) < LooseVersion('5.0'):
    libname = 'libgmx*.%s' % libext
else:
    libname = 'libgromacs*.%s' % libext
```

Comment (boegel) [inline easybuild/easyblocks/g/gromacs.py]: or even define `self.libext`, so you can use it in `sanity_check_step` too?

**Comment by gppezzi** [comment]
@boegel  any other remark or should I provide test reports for this version?
cc @lucamar

File: easybuild/easyblocks/g/gromacs.py [hunk 3]
```diff
@@ -41,7 +41,8 @@
 from easybuild.tools.build_log import EasyBuildError
 from easybuild.tools.modules import get_software_root
 from easybuild.tools.systemtools import get_platform_name
-
+from easybuild.tools.systemtools import get_shared_lib_ext
+import easybuild.tools.toolchain as toolchain
```
Comment (boegel) [inline easybuild/easyblocks/g/gromacs.py]: please move the `import` one up below the other `import` (keep them alphabetical), and collapse the two for `systemtools`:

```python
from easybuild.tools.systemtools import get_platform_name , get_shared_lib_ext
```

**Comment by boegel** [comment]
@gppezzi one more tiny remark, seeing test reports that rely on this would be great

**Comment by boegel** [comment]
@gppezzi I'm seeing this with existing easyconfigs:

```
ERROR: Traceback (most recent call last):
  File "/user/scratchdelcatty/gent/gvo000/gvo00002/vsc40023/easybuild_easy_installed/lib/python2.6/site-packages/easybuild_framework-2.8.1-py2.6.egg/easybuild/main.py", line 114, in build_and_install_software
    (ec_res['success'], app_log, err) = build_and_install_one(ec, init_env)
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_easy_installed/lib/python2.6/site-packages/easybuild_framework-2.8.1-py2.6.egg/easybuild/framework/easyblock.py", line 2358, in build_and_install_one
    result = app.run_all_steps(run_test_cases=run_test_cases)
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_easy_installed/lib/python2.6/site-packages/easybuild_framework-2.8.1-py2.6.egg/easybuild/framework/easyblock.py", line 2274, in run_all_steps
    self.run_step(step_name, step_methods)
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_easy_installed/lib/python2.6/site-packages/easybuild_framework-2.8.1-py2.6.egg/easybuild/framework/easyblock.py", line 2153, in run_step
    step_method(self)()
  File "/tmp/eb-Df_XrW/included-easyblocks/easybuild/easyblocks/gromacs.py", line 65, in configure_step
    if self.toolchain.options['dynamic']:
KeyError: 'dynamic'
```

So, you'll need to use

```python
self.toolchain.options.get('dynamic', False):
```

since the `dynamic` toolchain option may not be defined...

**Comment by boegel** [comment]
lgtm

there are a couple of other PRs for the GROMACS easyblock, I'll look into handling them together...

File: easybuild/easyblocks/g/gromacs.py [hunk 4]
```diff
@@ -97,7 +102,10 @@ def configure_step(self):
                 for libname in ['BLAS', 'LAPACK']:
                     lib_dir = os.getenv('%s_LIB_DIR' % libname)
                     libs = os.getenv('LIB%s' % libname)
-                    self.cfg.update('configopts', '-DGMX_%s_USER="-L%s %s"' % (libname, lib_dir, libs))
+                    if self.toolchain.toolchain_family() == toolchain.CRAYPE:
+                        self.cfg.update('configopts', '-DGMX_%s_USER="%s/libsci_gnu_mpi_mp.a"' % (libname, lib_dir))
```
Comment (boegel) [inline easybuild/easyblocks/g/gromacs.py]: hardcoding the `_gnu_` here means that this only works with `CrayGNU`

I've fixed this in #960 with a `glob`, cfr. ec6b342673ac92d75b734cacb1e116c1f35861d8

Comment (boegel) [inline easybuild/easyblocks/g/gromacs.py]: @gppezzi can you change this like I did in https://github.com/hpcugent/easybuild-easyblocks/pull/960/files#diff-a02b2451ce7221a206c1f64ba6758d33R158 (only the Cray-specific part in this `if` block), then I can merge this PR (and sync #960 after the merge)

**Comment by boegel** [comment]
retested with https://github.com/hpcugent/easybuild-easyblocks/pull/951 & all existing GROMACS easyconfigs using Cray toolchains, looks good, so going in

Thanks @gppezzi and @lucamar!

**Pull request merged**